tuf/client/client.go

```diff
@@ -439,18 +446,20 @@ func (c *Client) downloadTargets(role string) error {
 	return nil
 }
 
-func (c *Client) downloadSigned(role string, size int64, expectedSha256 []byte) ([]byte, *data.Signed, error) {
-	rolePath := utils.ConsistentName(role, expectedSha256)
+func (c *Client) downloadSigned(role string, size int64, expectedHashes data.Hashes) ([]byte, *data.Signed, error) {
+	// TODO:(hukeping) not sure about this, since we havn't do the sanity check of the expectedHashes yes.
```

Comment:
Yep, the existing implementation was discussed as part of the consistent download work. When downloading a timestamp, and in some instances a root, we will not have `expectedHashes`.

Also, consistent download is more prescriptive than other hash usages so it will only support sha256 for the foreseeable future, hence passing the one required hash, rather than the map.